integration/overrides_exporter_test.go

```diff
@@ -0,0 +1,154 @@
+			},
+			// the limit should be exported from both exporters, i.e. the expected sum is twice the limit value
+			expectedMetricSum: 2 * limit,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s, err := e2e.NewScenario(networkName)
+			require.NoError(t, err)
+			defer s.Close()
+
+			require.NoError(t, writeFileToSharedDir(
+				s,
+				overridesFileName,
+				[]byte(buildConfigFromTemplate(overridesFileContentTemplate, map[string]int{"MaxIngestionRate": limit})),
+			))
+
+			flags := map[string]string{
+				"-runtime-config.file": filepath.Join(e2e.ContainerSharedDir, overridesFileName),
+			}
+
+			services, err := newOverridesExporterServices(mergeFlags(flags, tt.flags), s)
+			require.NoError(t, err)
+
+			require.NoError(t, waitTenantSpecificMetricsAvailable(services.e1, services.e2, "tenant-a"))
+
+			value1, err := getOverrideMetricForTenantFromService("tenant-a", "ingestion_rate", services.e1)
+			require.NoError(t, err)
+			value2, err := getOverrideMetricForTenantFromService("tenant-a", "ingestion_rate", services.e2)
+			require.NoError(t, err)
+
+			require.Equal(t, tt.expectedMetricSum, int(value1+value2))
+		})
+	}
+}
+
+func waitTenantSpecificMetricsAvailable(s1, s2 *e2emimir.MimirService, tenantName string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	b := backoff.New(ctx, backoff.Config{
+		MinBackoff: 1 * time.Second,
+		MaxBackoff: 5 * time.Second,
+	})
+	for b.Ongoing() {
+		m1, err := s1.Metrics()
+		if err != nil {
+			return err
+		}
+		m2, err := s2.Metrics()
+		if err != nil {
+			return err
+		}
+		if strings.Contains(m1+m2, tenantName) {
+			return nil
+		}
+		b.Wait()
+	}
+	return errors.New("tenant specific metrics did not become available within timeout")
+}
+
+func getOverrideMetricForTenantFromService(tenantName string, limitName string, service *e2emimir.MimirService) (float64, error) {
+	sums, err := service.SumMetrics(
+		[]string{"cortex_limits_overrides"},
+		e2e.WithLabelMatchers(labels.MustNewMatcher(labels.MatchEqual, "user", tenantName)),
+		e2e.WithLabelMatchers(labels.MustNewMatcher(labels.MatchEqual, "limit_name", limitName)),
+		e2e.SkipMissingMetrics,
+	)
+	return sums[0], err
+}
+
+type overridesExporterServices struct {
+	e1 *e2emimir.MimirService
+	e2 *e2emimir.MimirService
+
+	// Dependencies
+	consul *e2e.HTTPService
+}
+
+func newOverridesExporterServices(flags map[string]string, s *e2e.Scenario) (*overridesExporterServices, error) {
+	consul := e2edb.NewConsul()
+	err := s.StartAndWaitReady(consul)
+	if err != nil {
+		return nil, err
+	}
+
+	exporter1 := e2emimir.NewOverridesExporter("overrides-exporter-1", consul.NetworkHTTPEndpoint(), flags)
+	err = s.StartAndWaitReady(exporter1)
+	if err != nil {
+		return nil, err
+	}
+	exporter2 := e2emimir.NewOverridesExporter("overrides-exporter-2", consul.NetworkHTTPEndpoint(), flags)
+	err = s.StartAndWaitReady(exporter2)
```

Comment:
It's slower to start the 1st service, then wait, then start the 2nd. It's better if you pass both services to the same `StartAndWaitReady()` call.